refactor: allow implicit allowed roles/locales (#314)
nhost/hasura-auth: .changeset/quiet-berries-develop.md

---
'hasura-auth': patch
---

> Always add required allowed roles and locales

Reviewer: ```suggestion
- The default role is automatically added to the allowed roles.
- The default locale is automatically added to the allowed locales.
```

> Until now, it was required to pass on explicilty the `"me"` and `AUTH_USER_DEFAULT_ROLE` to `AUTH_USER_DEFAULT_ALLOWED_ROLES`. They are now automatically added if they are not in the environment variable.

Reviewer: ```suggestion
Previously, it was explicitly required to add the `me` and `AUTH_USER_DEFAULT_ROLE` roles to `AUTH_USER_DEFAULT_ALLOWED_ROLES`. They are now automatically added to `AUTH_USER_DEFAULT_ALLOWED_ROLES`.
```

Before:

```
AUTH_USER_DEFAULT_ROLE=user
AUTH_USER_DEFAULT_ALLOWED_ROLES=user,me,other
```

Now, the following configuration will also work:

```
AUTH_USER_DEFAULT_ROLE=user
AUTH_USER_DEFAULT_ALLOWED_ROLES=other
```

> Both syntaxes will allow the roles `['user', 'me', 'other']`

Reviewer: ```suggestion
Both syntaxes will allow the roles `user`, `me`, and `other`.
```

> Similarly, it is not required to add the ``to the`AUTH_LOCALE_ALLOWED_LOCALES`.

Reviewer: ```suggestion
Similarly, it is no longer a requirement to add the value of `AUTH_LOCALE_DEFAULT` to the `AUTH_LOCALE_ALLOWED_LOCALES`.
```

Before:

```
AUTH_LOCALE_DEFAULT=en
AUTH_LOCALE_ALLOWED_LOCALES=en,fr
```

Now, the following configuration will also work:

```
AUTH_LOCALE_DEFAULT=en
AUTH_LOCALE_ALLOWED_LOCALES=en,fr
```

> Both syntwaxes will allow the locales `['en', 'fr']`

Reviewer: ```suggestion
Both syntaxes will allow the locales `en` and `fr`.
```
